File: app/flytec/lib/features/aplications_v2/models/dados_responsavel.dart
import 'dart:typed_data';

class DadosResponsavel {
  String? data;
  String? uf;
  String? cidade;
  String? nomeCompleto;
  String? documento;
  String? telefone;
  Uint8List? assinaturaResponsavel;

  DadosResponsavel({
    this.data,
    this.uf,
    this.cidade,
    this.nomeCompleto,
    this.documento,
    this.telefone,
    this.assinaturaResponsavel,
  });

  Map<String, dynamic> toMap() {
    return {
      'data': data,
      'uf': uf,
      'cidade': cidade,
      'nomeCompleto': nomeCompleto,
      'documento': documento,
      'telefone': telefone,
      'assinaturaResponsavel': assinaturaResponsavel,
    };
  }

  factory DadosResponsavel.fromJson(Map<String, dynamic>? json) {
    if(json==null) return DadosResponsavel();
    return DadosResponsavel(
      data: json['data'],
      uf: json['uf'],
      cidade: json['cidade'],
      nomeCompleto: json['nomeCompleto'],
      documento: json['documento'],
      telefone: json['telefone'],
      assinaturaResponsavel: json['assinaturaResponsavel'],
    );
  }
}

File: backoffice/Domain/Entidades/Cadastros/DadosResponsavel/DadosResponsavel.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace Domain.Entidades.Cadastros.DadosResponsavel;

public class DadosResponsavel
{
    [Key]
    public int Id { get; set; }
    public string? Data { get; set; }
    public string? UF { get; set; }
    public string? Cidade { get; set; }
    public string NomeCompleto { get; set; }
    public string? Documento { get; set; }
    public string? Telefone { get; set; }
    public string? PostoGraduacao { get; set; }
    public string? Re { get; set; }
    public byte[]? assinaturaResponsavel { get; set; }
    
    [ForeignKey("Empresa")]
    public int? IdEmpresa { get; set; }

    [JsonIgnore]
    public virtual Empresa.Empresa? Empresa { get; set; }
}
